_pages/exercises/Drones/package_delivery.md

```diff
@@ -0,0 +1,195 @@
+* `HAL.get_yaw_rate()` - Returns the actual yaw rate of the drone, in rad/s.
+* `HAL.get_orientation()` - Returns the actual roll, pitch and yaw of the drone as a numpy array [roll, pitch, yaw], in rad. 
+* `HAL.get_roll()` - Returns the roll angle of the drone, in rad
+* `HAL.get_pitch()` - Returns the pitch angle of the drone, in rad.
+* `HAL.get_yaw()` - Returns the yaw angle of the drone, in rad. 
+* `HAL.get_landed_state()` -  Returns 1 if the drone is on the ground (landed), 2 if the drone is in the air and 4 if the drone is landing. 0 could be also returned if the drone landed state is unknown. 
+* `HAL.get_pkg_state()` -  Returns True if the drone has picked up the package box else returns False.
+
+### Actuators and drone control
+
+The three following drone control functions are *non-blocking*, i.e. each time you send a new command to the aircraft it immediately discards the previous control command. 
+
+#### 1. Position control
+
+* `HAL.set_cmd_pos(x, y, z, az)` - Commands the *position* (x,y,z) of the drone, in m and the *yaw angle (az)* (in rad) taking as reference the first takeoff point (map frame)
+
+#### 2. Velocity control
+
+* `HAL.set_cmd_vel(vx, vy, vz, az)` - Commands the *linear velocity* of the drone in the x, y and z directions (in m/s) and the *yaw rate (az)* (rad/s) in its body fixed frame
+
+#### 3. Mixed control
+
+* `HAL.set_cmd_mix(vx, vy, z, az)` - Commands the *linear velocity* of the drone in the x, y directions (in m/s), the *height* (z) related to the takeoff point and the *yaw rate (az)* (in rad/s) 
+
+### Drone takeoff and land
+
+Besides using the buttons at the drone teleoperator GUI, taking off and landing can also be controlled from the following commands in your code:
+
+* `HAL.takeoff(height)` - Takeoff at the current location, to the given height (in m)
+* `HAL.land()` - Land at the current location. 
+
+### Drone cameras
+
+* `HAL.get_frontal_image()` - Returns the latest image from the frontal camera as a OpenCV cv2_image
+* `HAL.get_ventral_image()` - Returns the latest image from the ventral camera as a OpenCV cv2_image
+
+### Package box control
+
+* `HAL.set_cmd_pick()` - Drone picks up the package box from a minimum distance of 0.5 m
+* `HAL.set_cmd_drop()` - Drone drops the package box
+
+### GUI
+
+* `GUI.showImage(cv2_image)` - Shows a image of the camera  in the GUI
+* `GUI.showLeftImage(cv2_image)` - Shows another image of the camera in the GUI
+
+
+## Hints
+
+Simple hints provided to help you solve the package_delivery exercise. Please note that the **full solution has not been provided.**
+
+### Position Detection of the Target
+
+Notice that the target house has a visual signal near it. You can filter this signal in order to get the position of the target location.
+
+### How do I get the waypoint coordinates?
+
+1. **Green Pad:** Located at (x, y) = (-1, -4) m
+2. **Red Pad:** Located at (x, y) = (-1, -1) m
+2. **Target House:** Located at (x, y) = (20, 20) m
```

Comment:
Precise location? Which coordinate frame are you using? Are the pads pose needed?

Reply:
These positions are with respect to the coordinate frame used by Gazebo (fixed ENU frame).